### PR TITLE
Implement Subnet Resource

### PR DESCRIPTION
**Pull request by NavisJ**
# Description
Implement Subnet Resource

# ISSUE TYPE
- Feature Pull Request

##### RESOURCE NAME
Subnet

##### OUTPUT

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

File: examples/resources/powerscale_subnet/resource.tf [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+# Available actions: Create, Update, Delete and Import
+# After `terraform apply` of this example file it will create a new subnet with the name set in `name` attribute on the PowerScale
+
+# PowerScale subnet allows you to manage the subnet on the Powerscale array
+resource "powerscale_subnet" "subnet" {
```
Comment (RayLiu7) [inline examples/resources/powerscale_subnet/resource.tf:22]: Hi @NavisJ ,could you add optional fields for creating and updating in the example file as well?

File: examples/resources/powerscale_subnet/import.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# The command is
+# terraform import powerscale_subnet.subnet <id>
```
Comment (taohe1012) [inline examples/resources/powerscale_subnet/import.sh:17]: may change the <id> to <groupnet_id>.<subnet_id> ?

Comment (NavisJ) [reply examples/resources/powerscale_subnet/import.sh:17]: Good catch!

**Comment by NavisJ** [comment]
run e2e test

**Comment by forrestxia** [comment]
@NavisJ go generate failed

**Comment by NavisJ** [comment]
run e2e test

**Comment by NavisJ** [comment]
> @NavisJ go generate failed

fixed

**Comment by forrestxia** [comment]
run e2e test

**Pull request merged**